integration-libs/opf/base/components/opf-cta/opf-cta-scripts/opf-cta-scripts.service.ts

```diff
@@ -0,0 +1,245 @@
+          ctaProductItems: [{ productId: product?.code, quantity: 1 }],
+          paymentAccountIds: paymentAccountIds,
+          scriptLocations: [scriptLocation],
+        } as CtaScriptsRequest;
+      })
+    );
+  }
+
+  protected runCtaScripts(scripts: OpfDynamicScript[]) {
+    return from(scripts).pipe(
+      concatMap((script) => from(this.loadAndRunScript(script))),
+      reduce((loadedList: string[], script) => {
+        if (script?.html) {
+          loadedList.push(script.html);
+        }
+        return loadedList;
+      }, []),
+      map((list) => {
+        return this.removeScriptTags(list);
+      })
+    );
+  }
+
+  protected getScriptLocation(): Observable<CtaScriptsLocation | null> {
+    return this.cmsService.getCurrentPage().pipe(
+      take(1),
+      map((page) => {
+        switch (page.pageId) {
+          case CmsPageLocation.ORDER_PAGE:
+            return CtaScriptsLocation.ORDER_HISTORY_PAYMENT_GUIDE;
+          case CmsPageLocation.ORDER_CONFIRMATION_PAGE:
+            return CtaScriptsLocation.ORDER_CONFIRMATION_PAYMENT_GUIDE;
+          case CmsPageLocation.PDP_PAGE:
+            return CtaScriptsLocation.PDP_QUICK_BUY;
+          default:
+            return null;
+        }
+      })
+    );
+  }
+
+  protected getOrderDetails(scriptsLocation: CtaScriptsLocation) {
+    const order$ =
+      scriptsLocation === CtaScriptsLocation.ORDER_CONFIRMATION_PAYMENT_GUIDE
+        ? this.orderDetailsService.getOrderDetails()
+        : this.orderHistoryService.getOrderDetails();
+    return order$.pipe(
+      filter((order) => !!order?.entries)
+    ) as Observable<Order>;
+  }
+
+  protected getPaymentAccountIds() {
+    return this.opfPaymentFacade.getActiveConfigurationsState().pipe(
+      filter(
+        (state) => !state.loading && !state.error && Boolean(state.data?.length)
+      ),
+      map((state) => state.data?.map((val) => val.id) as number[])
+    );
+  }
+
+  protected getProductItems(
+    order: Order
+  ): { productId: string; quantity: number }[] | [] {
+    return (order.entries as OrderEntry[])
+      .filter((item) => {
+        return !!item?.product?.code && !!item?.quantity;
+      })
+      .map((item) => {
+        return {
+          productId: item.product?.code as string,
+          quantity: item.quantity as number,
+        };
+      });
+  }
+
+  protected loadAndRunScript(
+    script: OpfDynamicScript
+  ): Promise<OpfDynamicScript | undefined> {
+    const html = script?.html;
+
+    return new Promise(
+      (resolve: (value: OpfDynamicScript | undefined) => void) => {
+        this.opfResourceLoaderService
+          .loadProviderResources(script.jsUrls, script.cssUrls)
+          .then(() => {
+            if (html) {
+              this.opfResourceLoaderService.executeScriptFromHtml(html);
+              resolve(script);
+            } else {
+              resolve(undefined);
+            }
+          })
+          .catch(() => {
+            resolve(undefined);
```

Comment:
All `resolve(undefined)` are use for the logic : if a resource file (either css or js) fails to load, don't display the html snippet in UI.